Env check

bnmajor opened the pull request:
Adds an environment check notebook that is designed to check the environment that the user is running in to make sure that all required dependencies and extensions are correctly installed. Ideally run first before any other notebooks to prevent common issues around dependencies and extension loading.

bnmajor committed:
DOC: Add an environment check notebook

This notebook is designed to make sure that all requirements are installed in
the current environment and all extensions have been loaded before the user
runs any of the other notebooks.

bnmajor committed:
ENH: Bump itkwidgets to v1.0a23

bnmajor committed:
ENH: Bump itk to v5.3.0

bnmajor committed:
DOC: Update README to explain the EnvironmentCheck notebook

thewtex reviewed (approved):
LGTM, thank you @bnmajor !

bnmajor committed:
ENH: Generalize the install function to accept pre-release option

bnmajor committed:
DOC: Add warning about possible pip version resolution conflicts

Adds some explination as well as documentation for setting up a clean
environment and the option to open an issue if there are actual issues
running the notebook (or the others in the repo).

bnmajor committed:
BUG: Prevent notebook from getting caught in a refresh loop

The JS output is automatically evaluated when the notebook is opened. Make sure
the condition to force a refresh is always reset to false to prevent getting
stuck in an infinite loop of forcibly refreshing the page.

bnmajor commented:
Some additional updates:

- Allow the requirements list to include whether or not it should use the pre-release. Expects the following format:
    ```
    "{package_name}{(==|>=)version_number}{ --pre}"
    ```
    Where the package name is required but the version is optional, as is the `--pre` flag (expected at the end if included)
- Warn users that pip occasionally will raise errors when it is unable to resolve dependency conflicts. This can usually be safely ignored but there are now links to instructions on how to create a new, clean environment (either virtual env or conda) as well as a link to opening an issue if there are actual issues running notebooks in the repo after using the env check notebook.
- Prevent a bug that could possibly force the notebook to refresh non-stop
